transform/convert_quadkey_and_Vertical_id.go

```diff
@@ -438,6 +440,320 @@ func ConvertExtendedSpatialIDsToQuadkeysAndAltitudekeys(extendedSpatialIDs []str
+			for y := yMin; y <= yMax; y++ {
+				spatialIds = append(spatialIds, fmt.Sprintf("%v/%v/%v/%v", targetZoomLevel, extendedSpatialID.Z(), x, y))
+			}
+		}
+	case extendedSpatialID.HZoom() > extendedSpatialID.VZoom():
+		targetZoomLevel := extendedSpatialID.HZoom()
+		verticalIds := integrate.VerticalZoom(extendedSpatialID.VZoom(), extendedSpatialID.Z(), targetZoomLevel)
+		for _, verticalId := range verticalIds {
+			// "z/f" + "x/y"
+			spatialIds = append(spatialIds, fmt.Sprintf("%v/%v/%v", verticalId, extendedSpatialID.X(), extendedSpatialID.Y()))
+		}
+	// ズームレベルが等しい場合は直接変換可
+	case extendedSpatialID.HZoom() == extendedSpatialID.VZoom():
+		spatialId := fmt.Sprintf("%v/%v/%v/%v", extendedSpatialID.HZoom(), extendedSpatialID.Z(), extendedSpatialID.X(), extendedSpatialID.Y())
+		spatialIds = append(spatialIds, spatialId)
+	}
+	return spatialIds
+}
+
+// ConvertTileXYZsToExtendedSpatialIDs
+// TileXYZ形式から拡張空間ID形式へ変換する
+//
+// TileXYZのx,yは同一の意味のまま拡張空間IDのX,Yに変換される。
+// このため出力の水平精度は入力のものが用いられる(ただし精度チェックは行われる)。
+// TileXYZのzから拡張空間ID垂直インデックス(f)へは高度変換が用いられ変化する。
+// 引数 :
+//
+//	request : 変換対象のTileXYZ構造体のスライス
+//	zBaseExponent： TileXYZのzの高さが1mとなるズームレベル
+//	zBaseOffset： ズームレベルがzBaseExponentのとき高度0mにおけるTileXYZのz
+//	outputVZoom : 入力値が変換後の拡張空間IDの高さの精度となる。拡張空間IDの精度の閾値である 0 ～ 35 の整数値を指定可能。
+//
+// 戻り値 :
+//
+//	変換後の拡張空間IDのスライス
+//
+// 戻り値(エラー) :
+//
+//	以下の条件に当てはまる場合、エラーインスタンスが返却される。
+//	 精度閾値超過(出力精度)：出力の水平方向精度、または高さ方向の精度に 0 ～ 35 の整数値以外が入力されていた場合。
+//	 z高度範囲外：変換前のzがその垂直ズームレベルにおける高度範囲外である場合。
+//	 拡張空間ID高度範囲外：変換後の拡張空間ID高度がその垂直ズームレベルにおける高度範囲外である場合。
+//
+// 補足事項：
+//
+//	高さについて：
+//	 TileXYZ形式のzと拡張空間ID形式垂直インデックスは高度基準が異なる(同じにすることも可能)
+//	 引数のzBaseExponentとzBaseOffsetで高度基準を定義する
+//	 TileXYZ内のデータもしくは引数が次の状態のとき、入力TileXYZ数に対して出力拡張空間ID垂直インデックス数が増加する
+//	 - vZoomがzBaseExponentまたは25(空間IDの高度基準におけるzBaseExponent)より大きい場合
+//	 - zBaseOffsetが2のべき乗でない場合
+//
+// 変換利用例：
+//
+// 1. 入力TileXYZのzが出力拡張空間ID垂直インデックスに対応する場合
+//
+//	[]TileXYZ{
+//	    {
+//	        hZoom : 20
+//	        x 85263
+//	        y 65423
+//	        vZoom 23
+//	        z 0
+//	    }
+//	},
+//	zBaseExponent 25,
+//	zBaseOffset 8,
+//	outputVZoom 23
+//
+//	extendedSpatialIDs :["20/85263/65423/23/-2"]
+//
+// 2. vZoomが25より大きい場合
+//
+//	[]TileXYZ{
+//	    {
+//	        hZoom : 20
+//	        x 85263
+//	        y 65423
+//	        vZoom 26
+//	        z 3
+//	    }
+//	},
+//	zBaseExponent 25,
+//	zBaseOffset -2,
+//	outputVZoom 26
+//
+//	extendedSpatialIDs :["20/85263/65423/26/7", "20/85263/65423/26/7]
+//
+// 3. zBaseOffsetが2のべき乗でない場合
+//
+//	[]TileXYZ{
+//	    {
+//	        hZoom : 20
+//	        x 85263
+//	        y 65423
+//	        vZoom 23
+//	        z 0
+//	    }
+//	},
+//	zBaseExponent 25,
+//	zBaseOffset 7,
+//	outputVZoom 23
+//
+//	extendedSpatialIDs :["20/85263/65423/23/-2", "20/85263/65423/23/-1"]
+func ConvertTileXYZsToExtendedSpatialIDs(request []*object.TileXYZ, zBaseExponent uint16, zBaseOffset int64, outputVZoom uint16) ([]object.ExtendedSpatialID, error) {
```

Comment:
I want to argue whether to return []object.ExtendedSpatialID or []string .

Reply:
It is to be made using `[]object.ExtendedSpatialID` since this time